File: src/com/l2kt/gameserver/network/clientpackets/RequestQuestAbort.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.data.xml.ScriptData;
import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.scripting.Quest;
import com.l2kt.gameserver.scripting.QuestState;

public final class RequestQuestAbort extends L2GameClientPacket
{
	private int _questId;
	
	@Override
	protected void readImpl()
	{
		_questId = readD();
	}
	
	@Override
	protected void runImpl()
	{
		final Player activeChar = getClient().getActiveChar();
		if (activeChar == null)
			return;
		
		final Quest qe = ScriptData.getInstance().getQuest(_questId);
		if (qe == null)
			return;
		
		final QuestState qs = activeChar.getQuestState(qe.getName());
		if (qs != null)
			qs.exitQuest(true);
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/RequestQuestAbort.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.data.xml.ScriptData

class RequestQuestAbort : L2GameClientPacket() {
    private var _questId: Int = 0

    override fun readImpl() {
        _questId = readD()
    }

    override fun runImpl() {
        val activeChar = client.activeChar ?: return

        val qe = ScriptData.getQuest(_questId) ?: return

        val qs = activeChar.getQuestState(qe.name)
        qs?.exitQuest(true)
    }
}